uqcsbot/minecraft.py

```diff
@@ -93,6 +93,54 @@ async def mcwhitelist(self, interaction: discord.Interaction, username: str):
 
         db_session.close()
 
+    @app_commands.command()
+    @app_commands.describe(username="Minecraft username to unwhitelist.")
+    @yelling_exemptor(input_args=["username"])
+    async def mcunwhitelist(self, interaction: discord.Interaction, username: str):
+        """Removes a username from the whitelist for the UQCS server."""
+        db_session = self.bot.create_db_session()
+        query = db_session.query(MCWhitelist).filter(
+            MCWhitelist.discord_id == interaction.user.id
+        )
+        is_user_admin = (
+            isinstance(interaction.user, Member)
+            and interaction.user.guild_permissions.manage_guild
+        )
+
+        # If the user has already whitelisted someone, and they aren't an admin deny it.
+        if not is_user_admin and query.count() > 0:
+            await interaction.response.send_message(
+                "You've already whitelisted an account."
+            )
+        else:
+            # Send the RCON command to remove the user from the whitelist
+            response_remove = await self.send_rcon_command(
+                f"whitelist remove {username}"
+            )
+            logging.info(f"[MINECRAFT] whitelist remove {username}: {response_remove}")
+
+            # Send the RCON command to kick the player from the server
+            response_kick = await self.send_rcon_command(f"kick {username}")
+            logging.info(f"[MINECRAFT] kick {username}: {response_kick}")
+
+            # If the responses indicate successful removal and kick, remove the database item
+            if "Removed" in response_remove[0] and "Kicked" in response_kick[0]:
+                query.delete()
+                db_session.commit()
+
+                await self.bot.admin_alert(
+                    title="Minecraft Server Unwhitelist",
+                    description=f"{response_remove[0]}\n{response_kick[0]}",
+                    footer=f"Action performed by {interaction.user}",
+                    colour=Colour.red(),  # Use red to indicate removal
+                )
+
+            await interaction.response.send_message(
```

Comment:
This message could be beefed up a little bit, because it's realistically an error message - it couldn't kick or couldn't un-whitelist the player for some reason.

Reply:
Confused about this, lines 135 to 138 will just send a message to notify an individual has been kicked and unwhitelisted. 

<img width="317" alt="error" src="https://github.com/UQComputingSociety/uqcsbot-discord/assets/108224888/d93c167d-3f73-46aa-988b-27c96becb3a3">

Reply:
Hm, right. It was outside the check on line 127, which made me think it was an error case, but on second look it is just the normal message.

In that case, consider this a request for the bot to say/do something specific *if* the line 127 check fails, so we know if the bot fails to kick/fails to unwhitelist.

Reply:
After a bit of testing, I've just removed it and get it to display the remove_response[0] since removing a non-existence user will display results shown below:
![image](https://github.com/UQComputingSociety/uqcsbot-discord/assets/108224888/70a1ed05-8ff8-4cf3-a5f7-247ef7423916)